### PR TITLE
linkis-cli - dev-1.0.0

### PR DESCRIPTION
**Pull request by Fuu3214**
uploaded linkis-cli
changed directory of linkis-computation-client
some changes in gitignore

### CODE REVIEW

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/pom.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        <dependency>
+            <groupId>com.webank.wedatasphere.linkis</groupId>
+            <artifactId>linkis-cli-core</artifactId>
+            <version>${linkis.version}</version>
+        </dependency>
+        <!--<dependency>-->
+        <!--<groupId>com.webank.wedatasphere.linkis</groupId>-->
+        <!--<artifactId>linkis-ujes-client</artifactId>-->
+        <!--<version>${linkis.version}</version>-->
+        <!--</dependency>-->
```
Comment (Alexkun) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/pom.xml:32]: You can remove this if not needed.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/pom.xml [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
```
Comment (Alexkun) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/pom.xml:6]: License needed here.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/src/main/java/com/webank/wedatasphere/linkis/cli/application/driver/UjesClientDriver.java [hunk 3]
```diff
@@ -0,0 +1,595 @@
...
+                        builder.addServerUrl(ujesContext.getGatewayUrl())  //指定ServerUrl，linkis服务器端网关的地址,如http://{ip}:{port}
+                                .connectionTimeout(30000)   //connectionTimeOut 客户端连接超时时间
+                                .discoveryEnabled(false).discoveryFrequency(1, TimeUnit.MINUTES)  //是否启用注册发现，如果启用，会自动发现新启动的Gateway
+                                .loadbalancerEnabled(true)  // 是否启用负载均衡，如果不启用注册发现，则负载均衡没有意义
+                                .maxConnectionSize(5)   //指定最大连接数，即最大并发数
+                                .retryEnabled(false).readTimeout(ujesContext.getReadTimeoutMills())   //执行失败，是否允许重试
+                                .setAuthenticationStrategy(authenticationStrategy)   //AuthenticationStrategy Linkis认证方式
+                                .setAuthTokenKey(ujesContext.getTokenKey()).setAuthTokenValue(ujesContext.getTokenValue())))  //认证key，一般为用户名;  认证value，一般为用户名对应的密码
+                        .setDWSVersion(ujesContext.getDwsVersion()).build();  //linkis后台协议的版本，当前版本为v1
+
```
Comment (Alexkun) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-application/src/main/java/com/webank/wedatasphere/linkis/cli/application/driver/UjesClientDriver.java:94]: Please translate comments into English. And you can add Chinese comments additionally in complex parts.

**Review by yangzhiyue — APPROVED** [review]
thanks for this pr

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-common/src/main/java/com/webank/wedatasphere/linkis/cli/common/utils/GenericsUtils.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+ */
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+public class GenericsUtils {
+    /**
+     * 通过反射,获得定义Class时声明的父类的范型参数的类型. 如public BookManager extends
+     * GenricManager<Book>
+     *
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-common/src/main/java/com/webank/wedatasphere/linkis/cli/common/utils/GenericsUtils.java:29]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/command/fitter/SingleTplFitterTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * SingleTplFitter Tester.
+ *
+ * @author <Authors name>
+ * @version 1.0
+ * @since <pre>���� 14, 2021</pre>
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/command/fitter/SingleTplFitterTest.java:28]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/command/parser/SingleCmdParserTest.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * SingleCmdParser Tester.
+ *
+ * @author <Authors name>
+ * @version 1.0
+ * @since <pre>三月 14, 2021</pre>
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/command/parser/SingleCmdParserTest.java:35]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/properties/StdPropsLoaderTest.java [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+import org.junit.Test;
+
+import java.util.List;
+
+/**
+ * StdPropsLoader Tester.
+ *
+ * @author <Authors name>
+ * @version 1.0
+ * @since <pre>三月 14, 2021</pre>
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/properties/StdPropsLoaderTest.java:30]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/var/StdVarAccessTest.java [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * StdVarAccess Tester.
+ *
+ * @author <Authors name>
+ * @version 1.0
+ * @since <pre>三月 14, 2021</pre>
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/java/com/webank/wedatasphere/linkis/cli/core/interactor/var/StdVarAccessTest.java:47]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/resources/log4j2.xml [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<!--日志级别以及优先级排序: OFF > FATAL > ERROR > WARN > INFO > DEBUG > TRACE > ALL -->
+<!--Configuration后面的status，这个用于设置log4j2自身内部的信息输出，可以不设置，当设置成trace时，你会看到log4j2内部各种详细输出-->
+<!--monitorInterval：Log4j能够自动检测修改配置 文件和重新配置本身，设置间隔秒数-->
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/resources/log4j2.xml:17]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/resources/log4j2.xml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+        <!--filePattern="${log.path}/$${date:yyyy-linkis-clientient-%d{yyyy-MM-dd}-%i.log">-->
+        <!--<ThresholdFilter level="warn" onMatch="ACCEPT" onMismatch="DENY"/>-->
+        <!--<PatternLayout pattern="[%d{HH:mm:ss:SSS}] [%p] - %l - %m%n"/>-->
+        <!--<Policies>-->
+        <!--<TimeBasedTriggeringPolicy/>-->
+        <!--<SizeBasedTriggeringPolicy size="100 MB"/>-->
+        <!--</Policies>-->
+        <!--&lt;!&ndash; DefaultRolloverStrategy属性如不设置，则默认为最多同一文件夹下7个文件，这里设置了20 &ndash;&gt;-->
+        <!--<DefaultRolloverStrategy max="30"/>-->
+        <!--</RollingFile>-->
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/test/resources/log4j2.xml:41]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/pom.xml [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>linkis</artifactId>
+        <groupId>com.webank.wedatasphere.linkis</groupId>
+        <version>1.0.0-RC1</version>
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/pom.xml:26]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/pom.xml [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-api</artifactId>
+            <version>${log4j2.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-core</artifactId>
+            <version>${log4j2.version}</version>
+        </dependency>
+        <!-- 用于slf4j与log4j2桥接 -->
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-cli/pom.xml:65]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/request/JobListAction.scala [hunk 13]
```diff
@@ -0,0 +1,106 @@
+package com.webank.wedatasphere.linkis.ujes.client.request
+
+import com.webank.wedatasphere.linkis.httpclient.request.GetAction
+import org.apache.commons.lang.StringUtils
+
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * @author alexyang
+ * @date 2020/11/24
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/request/JobListAction.scala:10]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/request/OpenLogAction.scala [hunk 14]
```diff
@@ -0,0 +1,43 @@
+package com.webank.wedatasphere.linkis.ujes.client.request
+
+import com.webank.wedatasphere.linkis.httpclient.request.GetAction
+
+
+/**
+ * @author alexyang
+ * @date 2020/12/21
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/request/OpenLogAction.scala:8]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/response/JobListResult.scala [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+import java.util
+
+import com.webank.wedatasphere.linkis.httpclient.dws.annotation.DWSHttpMessageResult
+import com.webank.wedatasphere.linkis.httpclient.dws.response.DWSResult
+
+import scala.beans.BeanProperty
+
+/**
+ * @author alexyang
+ * @date 2020/11/24
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/response/JobListResult.scala:12]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/response/OpenLogResult.scala [hunk 16]
```diff
@@ -0,0 +1,25 @@
+package com.webank.wedatasphere.linkis.ujes.client.response
+
+import com.webank.wedatasphere.linkis.httpclient.dws.annotation.DWSHttpMessageResult
+import com.webank.wedatasphere.linkis.httpclient.dws.response.DWSResult
+
+import scala.beans.BeanProperty
+
+/**
+ * @author alexyang
+ * @date 2020/12/21
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/main/scala/com/webank/wedatasphere/linkis/ujes/client/response/OpenLogResult.scala:10]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+      .retryEnabled(false).readTimeout(30000) //执行失败，是否允许重试
+      .setAuthenticationStrategy(new StaticAuthenticationStrategy()) //AuthenticationStrategy Linkis认证方式
+      .setAuthTokenKey("${username}").setAuthTokenValue("${password}") //认证key，一般为用户名;  认证value，一般为用户名对应的密码
+      .setDWSVersion("v1").build() //Linkis后台协议的版本，当前版本为v1
+
+    // 2. 通过DWSClientConfig获取一个UJESClient
+    val client = UJESClient(clientConfig)
+
+    try {
+      // 3. 开始执行代码
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala:42]: Please modify it.

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-common/pom.xml [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>com.webank.wedatasphere.linkis</groupId>
+        <artifactId>linkis-cli</artifactId>
+        <version>1.0.0-RC1</version>
```
Comment (Alexkun) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-common/pom.xml:26]: Version should be 1.0.0

File: linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/main/java/com/webank/wedatasphere/linkis/cli/core/presenter/display/PlainTextFileDriver.java [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+        }
+
+        FileOutputStream fos = null;
+        OutputStreamWriter osWritter = null;
+        BufferedWriter bufferedWriter = null;
+        try {
+            fos = new FileOutputStream(file, !overWrite);
+            osWritter = new OutputStreamWriter(fos, "UTF-8");//设置字符编码
+            bufferedWriter = new BufferedWriter(osWritter, 1024);//设置缓冲区大小
+            bufferedWriter.write(content + "\n");
```
Comment (Alexkun) [inline linkis-computation-governance/linkis-client/linkis-cli/linkis-cli-core/src/main/java/com/webank/wedatasphere/linkis/cli/core/presenter/display/PlainTextFileDriver.java:67]: English comments here.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala [hunk 20]
```diff
@@ -0,0 +1,101 @@
...
+      .addServerUrl("http://${ip}:${port}") //指定ServerUrl，Linkis服务器端网关的地址,如http://{ip}:{port}
+      .connectionTimeout(30000) //connectionTimeOut 客户端连接超时时间
+      .discoveryEnabled(false).discoveryFrequency(1, TimeUnit.MINUTES) //是否启用注册发现，如果启用，会自动发现新启动的Gateway
+      .loadbalancerEnabled(true) // 是否启用负载均衡，如果不启用注册发现，则负载均衡没有意义
+      .maxConnectionSize(5) //指定最大连接数，即最大并发数
+      .retryEnabled(false).readTimeout(30000) //执行失败，是否允许重试
+      .setAuthenticationStrategy(new StaticAuthenticationStrategy()) //AuthenticationStrategy Linkis认证方式
+      .setAuthTokenKey("${username}").setAuthTokenValue("${password}") //认证key，一般为用户名;  认证value，一般为用户名对应的密码
+      .setDWSVersion("v1").build() //Linkis后台协议的版本，当前版本为v1
+
```
Comment (Alexkun) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala:37]: Please add English comments here.

File: linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala [hunk 21]
```diff
@@ -0,0 +1,101 @@
...
+
+object UJESClientImplBigResultSetDemoNew {
+
+
+  def main(args: Array[String]): Unit = {
+    val executeCode = "show tables"
+    val user = "hadoop"
+
+    // 1. 配置DWSClientBuilder，通过DWSClientBuilder获取一个DWSClientConfig
```
Comment (wushengyeyouya) [inline linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala:26]: Please modify all of Chinese to English.

Comment (Fuu3214) [reply linkis-computation-governance/linkis-client/linkis-computation-client/src/test/scala/com/webank/wedatasphere/linkis/ujes/UJESClientImplBigResultSetDemoNew.scala]: done, thx

**Review by wushengyeyouya — APPROVED** [review]
LGTM.

**Pull request merged**